app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/NrsService.scala

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.excisemovementcontrolsystemapi.services
+
+import play.api.Logging
+import play.api.http.Status.INTERNAL_SERVER_ERROR
+import uk.gov.hmrc.auth.core.authorise.EmptyPredicate
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals
+import uk.gov.hmrc.auth.core.retrieve._
+import uk.gov.hmrc.auth.core.{AffinityGroup, AuthConnector, ConfidenceLevel, CredentialRole}
+import uk.gov.hmrc.excisemovementcontrolsystemapi.connectors.NrsConnector
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.auth.ValidatedXmlRequest
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.nrs.{IdentityData, NonRepudiationSubmissionAccepted, NrsMetadata, NrsPayload}
+import uk.gov.hmrc.excisemovementcontrolsystemapi.services.NrsService.nonRepudiationIdentityRetrievals
+import uk.gov.hmrc.excisemovementcontrolsystemapi.utils.{DateTimeService, EmcsUtils, ErnsMapper, NrsEventIdMapper}
+import uk.gov.hmrc.http.{Authorization, HeaderCarrier, InternalServerException}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+
+@Singleton
+class NrsService @Inject()
+(
+  authConnector: AuthConnector,
+  nrsConnector: NrsConnector,
+  dateTimeService: DateTimeService,
+  emcsUtils: EmcsUtils,
+  ernsMapper: ErnsMapper,
+  nrsEventIdMapper: NrsEventIdMapper
+)(implicit ec: ExecutionContext) extends Logging {
+
+  def submitNrs(
+    request: ValidatedXmlRequest[_],
+  )(implicit headerCarrier: HeaderCarrier): Future[Either[Int, NonRepudiationSubmissionAccepted]] = {
+
+    val payload = request.body.toString
+    val userHeaderData = request.headersAsMap
+    val message = request.message
+    val exciseNumber = ernsMapper.getSingleErnFromMessage(message, request.validErns)
+    val notableEventId = nrsEventIdMapper.mapMessageToEventId(message)
+
+    (for {
+      identityData <- retrieveIdentityData()
+      userAuthToken = retrieveUserAuthToken(headerCarrier)
```

Comment:
We don't retrieve the API Key from the user's token do we? Don't we just need to add the Authorization token to the NRS call similar to how we do it for EIS?

Reply:
actually I am reviewing this

Reply:
yes, that is different from the API-Key which we need to get from NRS